lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +434,1140 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+              // Marker can be erased now, no longer needed.
+              Call->eraseFromParent();
+
+              auto Destination = getBasicBlockPC(Succ);
+              auto Edge = makeEdge(Destination,
+                                   model::FunctionEdgeType::FakeFunctionCall);
+              Block.Successors.insert(Edge);
+            } else {
+              Queue.insert(Succ);
+            }
+          }
+        }
+      }
+
+      CFG.insert(Block);
+    }
+  }
+
+  return CFG;
+}
+
+template<class FO>
+FunctionSummary
+CFEPAnalyzer<FO>::analyze(ArrayRef<GlobalVariable *> ABIRegisters,
+                          BasicBlock *Entry) {
+  using namespace llvm;
+
+  IRBuilder<> Builder(M.getContext());
+  StructType *MetaAddressTy = MetaAddress::getStruct(&M);
+
+  // Run function boundaries detection, and retrieve the outlined function.
+  Function *OutlinedFunction = createDisposableFunction(Entry);
+
+  // Initialize CFG of the function for the model.
+  auto CFG = collectDirectCFG(OutlinedFunction);
+
+  SmallVector<Instruction *, 4> IndirectBranchPredecessors;
+  if (AnyPCCloned) {
+    for (auto *Pred : predecessors(AnyPCCloned)) {
+      auto *Term = Pred->getTerminator();
+      IndirectBranchPredecessors.emplace_back(Term);
+    }
+  }
+
+  // Initialize ret-hook marker (needed for the ABI analyses on return values)
+  // and fix pre-hook marker upon encountering a jump to `anypc`. We don't know
+  // in advance whether it will be classified as a return or indirect tail call.
+  SmallVector<Instruction *, 4> BranchesForIBI;
+  for (auto *Term : IndirectBranchPredecessors) {
+    auto *BB = Term->getParent();
+    auto [RetPC, _] = getPC(Term);
+
+    auto CallSiteJT = GCBI->getJumpTarget(BB);
+    revng_assert(CallSiteJT.isValid());
+
+    auto *Split = BB->splitBasicBlock(Term, BB->getName() + Twine("__anypc"));
+    auto *JumpToAnyPC = Split->getTerminator();
+    revng_assert(isa<BranchInst>(JumpToAnyPC));
+    Builder.SetInsertPoint(JumpToAnyPC);
+
+    // JT of the call-site and unknown callee as arguments to pre hook.
+    Builder.CreateCall(PreHookMarker,
+                       { CallSiteJT.toConstant(MetaAddressTy),
+                         MetaAddress::invalid().toConstant(MetaAddressTy) });
+
+    Builder.CreateCall(RetHookMarker,
+                       { RetPC.toConstant(MetaAddressTy),
+                         MetaAddress::invalid().toConstant(MetaAddressTy) });
+
+    BranchesForIBI.emplace_back(JumpToAnyPC);
+  }
+
+  // Find registers that may be target of at least one store. This will help
+  // refine the final results.
+  std::set<GlobalVariable *> WrittenRegisters;
+  for (auto &BB : *OutlinedFunction) {
+    for (auto &I : BB) {
+      if (auto *SI = dyn_cast<StoreInst>(&I)) {
+        Value *Ptr = skipCasts(SI->getPointerOperand());
+        if (auto *GV = dyn_cast<GlobalVariable>(Ptr))
+          WrittenRegisters.insert(GV);
+      }
+    }
+  }
```

Comment:
Changed to:
```cpp
  for (auto &BB : *OutlinedFunction)
    for (auto &I : BB)
      if (auto *SI = dyn_cast<StoreInst>(&I)) {
        Value *Pointer = skipCasts(SI->getPointerOperand());
        if (auto *GV = dyn_cast<GlobalVariable>(Pointer))
          WrittenRegisters.insert(GV);
      }
```
since the last `if` did not pass `check-revng-conventions`.

Reply:
If a scope is delimited by `{ ... }` all the parent ones must be too.
Either revert to previous situation or maybe implement an helper function `isStoreToGlobal`.